index.bs

```diff
@@ -1043,6 +1045,608 @@ Note: Invocations of {{MLContext/compute()}} will fail if any of the {{MLContext
+    :: Support limits for operator instanceNormalization.
+    : <dfn>layerNormalization</dfn>
+    :: Support limits for operator layerNormalization.
+    : <dfn>leakyRelu</dfn>
+    :: Support limits for operator leakyRelu.
+    : <dfn>linear</dfn>
+    :: Support limits for operator linear.
+    : <dfn>lstm</dfn>
+    :: Support limits for operator lstm.
+    : <dfn>lstmCell</dfn>
+    :: Support limits for operator lstmCell.
+    : <dfn>matmul</dfn>
+    :: Support limits for operator matmul.
+    : <dfn>pad</dfn>
+    :: Support limits for operator pad.
+    : <dfn>averagePool2d</dfn>
+    :: Support limits for operator averagePool2d.
+    : <dfn>l2Pool2d</dfn>
+    :: Support limits for operator l2Pool2d.
+    : <dfn>maxPool2d</dfn>
+    :: Support limits for operator maxPool2d.
+    : <dfn>prelu</dfn>
+    :: Support limits for operator prelu.
+    : <dfn>reduceL1</dfn>
+    :: Support limits for operator reduceL1.
+    : <dfn>reduceL2</dfn>
+    :: Support limits for operator reduceL2.
+    : <dfn>reduceLogSum</dfn>
+    :: Support limits for operator reduceLogSum.
+    : <dfn>reduceLogSumExp</dfn>
+    :: Support limits for operator reduceLogSumExp.
+    : <dfn>reduceMax</dfn>
+    :: Support limits for operator reduceMax.
+    : <dfn>reduceMean</dfn>
+    :: Support limits for operator reduceMean.
+    : <dfn>reduceMin</dfn>
+    :: Support limits for operator reduceMin.
+    : <dfn>reduceProduct</dfn>
+    :: Support limits for operator reduceProduct.
+    : <dfn>reduceSum</dfn>
+    :: Support limits for operator reduceSum.
+    : <dfn>reduceSumSquare</dfn>
+    :: Support limits for operator reduceSumSquare.
+    : <dfn>relu</dfn>
+    :: Support limits for operator relu.
+    : <dfn>resample2d</dfn>
+    :: Support limits for operator resample2d.
+    : <dfn>reshape</dfn>
+    :: Support limits for operator reshape.
+    : <dfn>sigmoid</dfn>
+    :: Support limits for operator sigmoid.
+    : <dfn>slice</dfn>
+    :: Support limits for operator slice.
+    : <dfn>softmax</dfn>
+    :: Support limits for operator softmax.
+    : <dfn>softplus</dfn>
+    :: Support limits for operator softplus.
+    : <dfn>softsign</dfn>
+    :: Support limits for operator softsign.
+    : <dfn>split</dfn>
+    :: Support limits for operator split.
+    : <dfn>tanh</dfn>
+    :: Support limits for operator tanh.
+    : <dfn>transpose</dfn>
+    :: Support limits for operator transpose.
+    : <dfn>triangular</dfn>
+    :: Support limits for operator triangular.
+    : <dfn>where</dfn>
+    :: Support limits for operator where.
+</dl>
+#### {{MLSupportLimits}} dictionary #### {#api-mlcontext-supportlimits-dictionary}
+<script type="idl">
+dictionary MLSupportLimits {
+  sequence<DOMString> dataTypes;
+};
+</script>
+<dl dfn-type=dict-member dfn-for=MLSupportLimits>
+    : <dfn>dataTypes</dfn>
+    :: Supported data types.
+</dl>
+#### {{MLBatchNormalizationSupportLimits}} dictionary #### {#api-mlcontext-batchnormalizationsupportlimits-dictionary}
```

Comment:
For readability, what do you think about moving this into the WebIDL section for [`batchNormalization()`](https://webmachinelearning.github.io/webnn/#api-mlgraphbuilder-batchnorm)?

This'll help break up what's otherwise a big pile of repetitive struct definitions and instead make it easier to cross reference between the arguments to a builder method and the support limits dictionary for it.

Reply:
Right now all the opSupportLimits is self contained within a section. If we move the MLBatchNormalizationSupportLimits to the batchNormalization() idl, I feel it's confusing to see the definition of MLBatchNormalizationSupportLimits that's not used by the rest of the idl script, same for the dictionary arguments. It just doesn't fit very well to the [batchNormalization()](https://webmachinelearning.github.io/webnn/#api-mlgraphbuilder-batchnorm) paragraphs?

Reply:
We could use `partial dictionary` definitions to put everything together with the definition of `batchNormalization()`:

```
partial dictionary MLOpSupportLimits {
  MLBatchNormalizationSupportLimits batchNormalization;
};

dictionary MLBatchNormalizationSupportLimits {
  MLSupportLimits input;
  MLSupportLimits mean;
  MLSupportLimits variance;
  MLSupportLimits scale;
  MLSupportLimits bias;
  MLSupportLimits output;
};
```

That way everything relevant to an operator is together in the section of the spec that defines the operator, and using a `partial dictionary` means that `MLBatchNormalizationSupportLimits` is used locally in the same section of WebIDL.

I feel like it should fit into the [batchNormalization()](https://webmachinelearning.github.io/webnn/#api-mlgraphbuilder-batchnorm) paragraphs since it's all describing information about the same op.

Reply:
Oh I see, yeah I can do that.